Snackbar design

kolanse opened the pull request:
- Added a snackbar to notify users when an alarm is set 
- Please refer to issue  #352 for more information
 

![WhatsApp Image 2021-07-22 at 22 37 09 (4)](https://user-images.githubusercontent.com/68715929/126720623-b3b1293c-5911-426c-9149-785c01735f8c.jpeg)

![WhatsApp Image 2021-07-22 at 22 37 09 (3)](https://user-images.githubusercontent.com/68715929/126720626-3246c942-5eb6-41fb-82ec-929cff9b1588.jpeg)

![WhatsApp Image 2021-07-22 at 22 37 09 (2)](https://user-images.githubusercontent.com/68715929/126720630-6095dd83-099a-4502-b930-ff5dbaa0ddd5.jpeg)

![WhatsApp Image 2021-07-22 at 22 37 09 (1)](https://user-images.githubusercontent.com/68715929/126720633-050fad71-4abb-41ff-9b94-020b5e3c9168.jpeg)

![WhatsApp Image 2021-07-22 at 22 37 09](https://user-images.githubusercontent.com/68715929/126720638-9b372a95-d348-482a-b45c-37b4f946a97e.jpeg)

yuriykulikov commented on app/src/main/java/com/better/alarm/presenter/AlarmsListActivity.kt:
```diff
@@ -189,13 +203,52 @@ class AlarmsListActivity : AppCompatActivity() {
+        store.sets().withLatestFrom<Boolean, Pair<Store.AlarmSet, Boolean>>(
+            store.uiVisible,
+            BiFunction { set, uiVisible -> set to uiVisible })
+            .subscribe { (set: Store.AlarmSet, uiVisible: Boolean) ->
```
Leaked disposable here. This subscription will not be disposed when the Fragment is destroyed

- [ ] Add disposable to the composite disposable of the fragment

kolanse replied:
This code is in the activity

kolanse replied:
Done . i added it in the onStop of the activity

yuriykulikov commented on app/src/main/java/com/better/alarm/presenter/AlarmsListActivity.kt:
```diff
@@ -189,13 +203,52 @@ class AlarmsListActivity : AppCompatActivity() {
+    }
+
+    private fun configureSnackbar() {
+        val rootView = window.decorView.rootView
```
Can this fail? Does it make sense to move it inside the subscribe block?

kolanse replied:
Yeah it does makes sense.

yuriykulikov commented on app/src/main/java/com/better/alarm/presenter/ToastPresenter.kt:
```diff
@@ -1,105 +1,58 @@
+import android.widget.Toast
+import com.better.alarm.R
+import com.better.alarm.configuration.Store
+import com.better.alarm.util.formatToast
```
These imports are not necessary here, right?

yuriykulikov commented on app/src/main/res/values-v21/themes_deep_blue.xml:
```diff
@@ -32,6 +36,16 @@
         <item name="drawerClosedBackgroundColor">@color/deepblue_background</item>
     </style>
+    
+    <style name="Widget.App.SnackbarDefaultDeepBlueTheme" parent="" >
```
Perhaps remove these empty lines?

yuriykulikov commented on app/src/main/java/com/better/alarm/presenter/ToastPresenter.kt:
```diff
@@ -1,105 +1,58 @@
+                    if (!uiVisible && set.alarm.isEnabled) {
+                        popAlarmSetToast(context, set.millis)
+                    }
+
```
Empty lines

yuriykulikov reviewed (changes requested):
Looks good to me, but please prepare the commit for merging by reverting the unnecessary empty lines and changes in unrelated files. Here is how you can do that, or at least this is how I do that :)

1. in the git console run `git reset HEAD^`
2. In the Android Studio go to the Commit tab and open all files which are changed one by one and check what has been changed
3. Revert changes in unrelated places, such as inserting empty lines and so on

Also, it would be nice if there was a test, but I guess it would be quite hard to make it reliable :(

kolanse committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

removed empty lines

# This is the commit message #2:

removed empty lines

kolanse commented:
@yuriykulikov  i have made the recommended changes and pushed. do i need to create a new PR request. Also for the test. do you have a test written for the toastpresenter class. perhaps it might serve as a guide for me

yuriykulikov commented:
@kolanse as you can see now github cannot merge the PR automatically.

Could you please do the following?

`git remote add upstream git@github.com:yuriykulikov/AlarmClock.git`

`git fetch upstream`

`git rebase upstream/redesign -i`

-squash everything- 

`git push origin HEAD -f`

These commands will update your PR, which will allow me to merge this.

As for the test, no I do not have a test for the toast thing :) If you don't have prior knowledge about espresso tests it will be very time consuming to implement a test for the snackbar. Let's merge this one first.

kolanse commented:
@yuriykulikov  Thank you very much. i will follow your instructions. as for the espresso test. i will need to time to work on it because it might be time-consuming. but as you said let us merge this first. I will work on it later given the chance

kolanse committed:
removed empty lines

removed empty lines

added snackbar design

final commit

kolanse commented:
@yuriykulikov  i have force pushed as instructed and done the other instructions. Thank you

yuriykulikov commented:
Merged. @kolanse thank you for the contribution!

kolanse commented:
@yuriykulikov  you are welcome
